worlds/tloz/ItemPool.py

```diff
@@ -87,8 +87,13 @@ def get_pool_core(world):
     minor_items = dict(minor_dungeon_items)
 
     # Guaranteed Shop Items
-    reserved_store_slots = random.sample(shop_locations[0:9], 4)
-    for location, item in zip(reserved_store_slots, guaranteed_shop_items):
+    arrow = "Arrow"
+    copy_shop_items = guaranteed_shop_items.copy()
+    if (world.options.ArrowLocation == ArrowLocation.option_anywhere) and (arrow in copy_shop_items):
+        copy_shop_items.remove(arrow)
+        pool.append(arrow)
+    reserved_store_slots = random.sample(shop_locations[0:9], len(copy_shop_items))
+    for location, item in zip(reserved_store_slots, copy_shop_items):
```

Comment:
I think that if we are going to go this route, Arrow should be removed from guaranteed_shop_items and then added to copy_shop_items only if ArrowLocation == shop. This will make things cleaner and easier for if/when I get around to expanding this option to allow other items to be in guaranteed_shop_items.

Reply:
Do the other three items currently guaranteed in shop have any logical ramifications? Because if not, it wouldn't be that hard to refactor this PR to enable full shop control -- just drop a List in the settings that can be customized and pulled.

Reply:
Regarding "Arrow defaults in shop" vs "Arrow defaults in pool" though, I agree. It was formatted this way in the troubleshooting process because this way worked and the other way didn't at first, due to an unrelated bug that has since been quashed. I'll commit another change to do this.